dmapiclient/search.py

```diff
@@ -142,7 +143,25 @@ def search_services_from_url(self, search_api_url, id_only=False, page=None):
 
         return self._get(paged_search_api_url)
 
-    search_services_from_url_iter = make_iter_method('search_services_from_url', 'services')
+    # search_services_from_url_iter = make_iter_method('search_services_from_url', 'documents')
+
+    # This is a temporary method to allow for backwards compatability while switching from the search-api returning
+    # `documents` instead of just `services`. The data-api uses this method when locking direct award searches. The api
+    # will need to pull in this apiclient before the search-api changes. Once it does, this method can be removed and
+    # the calle to `make_iter_method` above can be uncommented.
+    def search_services_from_url_iter(self, search_url, id_only):
+        backoff_decorator = backoff.on_exception(backoff.expo, HTTPTemporaryError, max_tries=5)
+        result = backoff_decorator(self.search_services_from_url)(search_url, id_only=id_only)
+        for service in result.get('documents') or result.get('services'):
```

Comment:
In the case where `documents` were a zero-length sequence, that being Falsey would cause it to attempt to `.get(` `services` which would of course be absent, leading the `for` loop to try and iterate through the `None` it got from that, raising an exception.

Could solve by adding an `or ()` to the end of the chain or modifying the last call to `result.get('services', ())`.

Reply:
(thanks though, this should make my job easier)

Reply:
Have removed this now and update `make_iter_method` to do something different. Would be good to get your eyes on it.